hedera-node/src/main/java/com/hedera/services/state/submerkle/ExpirableTxnRecord.java

```diff
@@ -913,55 +917,53 @@ public void excludeHbarChangesFrom(final ExpirableTxnRecord.Builder that) {
 			final var adjustsHere = this.transferList.hbars.length;
 			final var adjustsThere = that.transferList.hbars.length;
 			final var maxAdjusts = adjustsHere + adjustsThere;
-			final var changedHere = this.transferList.accountIds;
-			final var changedThere = that.transferList.accountIds;
+			final var changedHere = this.transferList.accountCodes;
+			final var changedThere = that.transferList.accountCodes;
+			final var maxAccountCodes = changedHere.length + changedThere.length;
+
 
 			final var netAdjustsHere = new long[maxAdjusts];
-			final List<EntityId> netChanged = new ArrayList<>();
+			final long[] netChanged = new long[maxAccountCodes];
 
 			var i = 0;
 			var j = 0;
 			var k = 0;
+			var m = 0;
```

Comment:
Imo it's slightly confusing to have a separate `m` tracker, since its value must always be identical to `k`. Maybe we can just `netAdjustsHere[k] = ...` then do `netChanged[k++] = ...` below?